### PR TITLE
E1689: Anonymous Chat Between Author and Reviewer

### PR DESCRIPTION
**Pull request by saratkavuru**
This is the version where the reviewee would be able to chat with all of their reviewers at once in a pop up box.

### CODE REVIEW

**Comment by yangsong8** [comment]
You submitted two pull requests, #880 and #853, which is the right one, plz?

**Comment by saratkavuru** [comment]
Hi Yang,
#853 is the version where each of  the reviewers have a separate chat with
the reviewee team.
#880 is the version where all of the reviewers have a common chat with the
reviewee team.
We have earlier done the #853 version and later after talking to the
professor , we have created #880 from another branch.
Please go through #880 first.

ᐧ

File: app/helpers/messages_helper.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module MessagesHelper
+
+def reviewee_or_reviewer(message)
```
Comment (yangsong8) [inline app/helpers/messages_helper.rb]: I don't like this method much. Better to have an 'is_reviewer' method?

File: app/helpers/messages_helper.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	team = AssignmentTeam.find(message.chat.assignment_team_id)
+  for user in team.users
+    if message.user==user
+      return "Reviewee"
+    end
+      end
+	return "Reviewer"
```
Comment (yangsong8) [inline app/helpers/messages_helper.rb]: White spaces don't look right to me.

File: app/helpers/messages_helper.rb (hunk 1 above)
Comment (saratkavuru) [inline app/helpers/messages_helper.rb]: @yangsong8  I have fixed this and changed the code to display Author instead of Reviewer in the chat box.

**Comment by yangsong8** [comment]
Hi guys, I got trouble in running your code locally. Below is the error message I got. 

image

Do you have any guess on what I miss? Thanks.

**Comment by saratkavuru** [comment]
Hi Yang, 
please run the following command **rails generate sync:install**
And then start the faye server by **rackup sync.ru -E production** along with **rails server**  in another shell.

**Comment by yangsong8** [comment]
Thank you for your response!

**Comment by saratkavuru** [comment]
Np, Did it solve the issue?

**Comment by yangsong8** [comment]
Yes, I am moving on to test your project further. Yours is a big project considering the number of lines. I will check it carefully before making a decision. Thanks again.

**Comment by yangsong8** [comment]
I got another issue. Below is what I have done.

- add a new participant.
- have the participant select a topic (thereby an assignment team is created and a chat object too)
- have a reviewer review the newly submitted work (chat object is created)

Now I am impersonating the review and try to say something. When I hit enter, I got an error on Rails console:
image

**Comment by saratkavuru** [comment]
Hi Yang, we missed this usage instance of and old method. I have changed the method to its new name. Sorry for the trouble.

**Comment by Winbobob** [comment]
I tried. The effect is different from your video. Each time after send a new message, I can see it only after reopen the chatting window.

**Pull request closed without merging**